core/dbt/parser/manifest.py

```diff
@@ -1598,6 +1598,44 @@ def _process_metric_node(
             )
             metric.type_params.input_measures.extend(target_metric.type_params.input_measures)
             metric.depends_on.add_node(target_metric.unique_id)
+    elif metric.type is MetricType.CONVERSION:
+        conversion_type_params = metric.type_params.conversion_type_params_or_error
+        metric.type_params.input_measures.append(conversion_type_params.base_measure)
+        metric.type_params.input_measures.append(conversion_type_params.conversion_measure)
+
+        target_base_semantic_model = manifest.resolve_semantic_model_for_measure(
+            target_measure_name=conversion_type_params.base_measure.name,
+            current_project=current_project,
+            node_package=metric.package_name,
+        )
+        if target_base_semantic_model is None:
+            raise dbt.exceptions.ParsingError(
+                f"A semantic model having a measure `{conversion_type_params.base_measure.name}` does not exist but was referenced.",
+                node=metric,
+            )
+        if target_base_semantic_model.config.enabled is False:
+            raise dbt.exceptions.ParsingError(
+                f"The measure `{conversion_type_params.base_measure.name}` is referenced on disabled semantic model `{target_base_semantic_model.name}`.",
+                node=metric,
+            )
+
+        target_conversion_semantic_model = manifest.resolve_semantic_model_for_measure(
+            target_measure_name=conversion_type_params.conversion_measure.name,
+            current_project=current_project,
+            node_package=metric.package_name,
+        )
+        if target_conversion_semantic_model is None:
+            raise dbt.exceptions.ParsingError(
+                f"A semantic model having a measure `{conversion_type_params.conversion_measure.name}` does not exist but was referenced.",
+                node=metric,
+            )
+        if target_conversion_semantic_model.config.enabled is False:
+            raise dbt.exceptions.ParsingError(
+                f"The measure `{conversion_type_params.conversion_measure.name}` is referenced on disabled semantic model `{target_conversion_semantic_model.name}`.",
+                node=metric,
+            )
+        metric.depends_on.add_node(target_base_semantic_model.unique_id)
+        metric.depends_on.add_node(target_conversion_semantic_model.unique_id)
```

Comment:
This looks to follow the same process as what we do for `MetricType.CUMULATIVE` and `MetricType.SIMPLE` metric types, except we also do it twice. Could we de-duplicate this? I think we could do so by at the beginning of what would be `if metric.type is MetricType.SIMPLE or metric.type is MetricType.CUMULATIVE or metric.type is MetricType.CONVERSION`
We first build the list of measure inputs to lookup (for simple / cumulative it'd be a list with 1 measure input, for conversion it'd be a list of two). Then we just do necessary logic in a loop.